src/anndata/_core/aligned_df.py

```diff
@@ -91,9 +91,9 @@ def _gen_dataframe_df(
     if not is_string_dtype(anno.index[~anno.index.isna()]):
         msg = "Transforming to str index."
         warn(msg, ImplicitModificationWarning)
-        anno.index = anno.index.astype("str")
+        anno.index = anno.index.astype(pandas_str_dtype())
```

Comment:
Presumably this line is problematic for the failing tests because (I think) they all involve converting a range index to an `object` type